### PR TITLE
openboardview: Init at 8.0

### PR DESCRIPTION
**Pull request by felixsinger**
###### Motivation for this change
Software for viewing PCB/Laptop/Motherboard Layouts.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/electronics/openboardviewer/default.nix [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    url = "https://raw.githubusercontent.com/KhronosGroup/OpenGL-Registry/e82287b71f66a88194f4a39e35fdac83dbf72707/xml/glx.xml";
+    sha256 = "4f00f20e507c353e8cc7765cd414b659a53628a12c5c22459afdc7ddd0f620b6";
+  };
+
+  src_wgl = fetchurl {
+    url = "https://raw.githubusercontent.com/KhronosGroup/OpenGL-Registry/e82287b71f66a88194f4a39e35fdac83dbf72707/xml/wgl.xml";
+    sha256 = "4f36192884ce3e95b3a27d4ae1763fa57aba677fd31b421f8c496140f17538b6";
+  };
+
+  buildInputs = with pkgs; [
```
Comment (siraben) [inline pkgs/applications/science/electronics/openboardviewer/default.nix:41]: Don't import the package set directly, import them one by one.

File: pkgs/applications/science/electronics/openboardviewer/default.nix [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    (SDL2.override { withStatic = true; }).dev
+    gnumake cmake python3Full pkg-config zlib
```
Comment (siraben) [inline pkgs/applications/science/electronics/openboardviewer/default.nix:43]: `cmake`, `pkg-config`, `gcc10`, should go in `nativeBuildInputs`. Roughly speaking, they are build-time rather than runtime dependencies.

File: pkgs/applications/science/electronics/openboardviewer/default.nix [hunk 3]
```diff
@@ -0,0 +1,62 @@
+{ stdenv, fetchurl, fetchFromGitHub, pkgs }:
```
Comment (siraben) [inline pkgs/applications/science/electronics/openboardviewer/default.nix:1]:
```suggestion
{ stdenv, lib, fetchurl, fetchFromGitHub, cmake, gnumake, pkg-config, zlib }:
```
And so on

File: pkgs/applications/science/electronics/openboardviewer/default.nix [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+  preBuild = ''
+    cp ${src_egl} ${glad_path}/egl.xml
+    cp ${src_gl}  ${glad_path}/gl.xml
+    cp ${src_glx} ${glad_path}/glx.xml
+    cp ${src_wgl} ${glad_path}/wgl.xml
+    cp ${src_khrplatform} ${glad_path}/khrplatform.h
+  '';
+
+  meta = with stdenv.lib; {
```
Comment (siraben) [inline pkgs/applications/science/electronics/openboardviewer/default.nix:55]:
```suggestion
  meta = with lib; {
```

File: maintainers/maintainer-list.nix [hunk 5]
```diff
@@ -3001,6 +3001,12 @@
     githubId = 9959940;
     name = "Andreas Fehn";
   };
+  felixsinger = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:3004]: Please put this into a separate commit with the message ``maintainers: add felixsinger``.

File: pkgs/applications/science/electronics/openboardviewer/default.nix [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+  src = fetchFromGitHub {
+    owner = "OpenBoardView";
+    repo = pname;
+    rev = "45edcee7e4c67cd90522d070431c920214002027";
+    sha256 = "1wxr8c8zyxxp4addww6ml5a0pyknvb4lph1hd0lhzx4asjwjnz46";
+    fetchSubmodules = true;
+  };
+
+  src_egl = fetchurl {
```
Comment (SuperSandro2000) [inline pkgs/applications/science/electronics/openboardviewer/default.nix:16]: Please move this into a let in.

File: maintainers/maintainer-list.nix (hunk 5 above)
Comment (felixsinger) [reply maintainers/maintainer-list.nix:3004]: Already done in commit 32836182e

File: pkgs/applications/science/electronics/openboardviewer/default.nix (hunk 2 above)
Comment (felixsinger) [reply pkgs/applications/science/electronics/openboardviewer/default.nix:47]: If I put these into nativeBuildInputs it doesn't build anymore.

Comment (siraben) [reply pkgs/applications/science/electronics/openboardviewer/default.nix:47]: Why not?

File: maintainers/maintainer-list.nix [hunk 7]
```diff
@@ -3001,6 +3001,12 @@
...
+    email = "felixsinger@posteo.net";
+    github = "felixsinger";
+    githubId = 628359;
+    name = "Felix Singer";
+  }
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:3009]:
```suggestion
  };
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by siraben** [comment]
@felixsinger please fix the evaluation error

File: pkgs/applications/science/electronics/openboardviewer/default.nix [hunk 8]
```diff
@@ -0,0 +1,66 @@
+{ stdenv, lib, fetchurl, fetchFromGitHub,
+  gnumake, cmake, python3Full, pkg-config,
+  zlib, sqlite, fontconfig, gtk3, gcc10,
+  SDL2 }:
```
Comment (SuperSandro2000) [inline pkgs/applications/science/electronics/openboardviewer/default.nix:4]:
```suggestion
{ stdenv, lib, fetchurl, fetchFromGitHub
, gnumake, cmake, python3Full, pkg-config
, zlib, sqlite, fontconfig, gtk3, gcc10
, SDL2 }:
```

File: pkgs/applications/science/electronics/openboardviewer/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/applications/science/electronics/openboardviewer/default.nix:47]: also gnumake should be available by default and also belongs into nativeBuildInputs.

Comment (felixsinger) [reply pkgs/applications/science/electronics/openboardviewer/default.nix:48]: At the moment, I am not able to build it anymore. I am getting the same errors as when I moved these packages to nativeBuildInputs. I will try to investigate that further.

```
/build/source/src/openboardview/FileFormats/BRDFile.cpp:216:37: error: expected unqualified-id before '&' token
  216 |    } catch (const std::out_of_range &e) {
      |                                     ^
/build/source/src/openboardview/FileFormats/BRDFile.cpp:216:36: error: expected ')' before '&' token
  216 |    } catch (const std::out_of_range &e) {
      |            ~                       ^~
      |                                    )
/build/source/src/openboardview/FileFormats/BRDFile.cpp:216:37: error: expected '{' before '&' token
  216 |    } catch (const std::out_of_range &e) {
      |                                     ^
/build/source/src/openboardview/FileFormats/BRDFile.cpp:216:38: error: 'e' was not declared in this scope
  216 |    } catch (const std::out_of_range &e) {
```

Comment (felixsinger) [reply pkgs/applications/science/electronics/openboardviewer/default.nix]: After some investigation, the issue was a too new GCC version. GCC 10 causes that error, GCC 9 works fine.
